components/service/glean/src/main/java/mozilla/components/service/glean/net/HttpPingUploader.kt

```diff
@@ -0,0 +1,100 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.service.glean.net
+
+import android.support.annotation.VisibleForTesting
+import mozilla.components.service.glean.config.Configuration
+import mozilla.components.support.base.log.logger.Logger
+import java.io.IOException
+import java.net.HttpURLConnection
+import java.net.MalformedURLException
+import java.net.URL
+import java.net.URLConnection
+
+/**
+ * A simple ping Uploader, which implements a "send once" policy, never
+ * storing or attempting to send the ping again.
+ */
+class HttpPingUploader(configuration: Configuration) : PingUploader {
+    private val config = configuration
+    private val logger = Logger("glean/HttpPingUploader")
+
+    override fun upload(path:String, data: String): Boolean {
+        try {
+            with(openConnection(config.serverEndpoint, path) as HttpURLConnection) {
```

Comment:
One thing that is missing here is that we never seem to call `disconnect()` on the connection. Usually that's part of the `finally` block of the `try` block to make sure it's always executed.